### PR TITLE
Blacklisting tests + DTM rewards

### PR DESCRIPTION
**Pull request by marcinwadon**

### CODE REVIEW

**Review by kpudlik — DISMISSED** [review]

File: src/main/scala/org/constellation/rewards/DTM.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.constellation.rewards
+
+trait DTM {
```
Comment (TheMMaciek) [inline src/main/scala/org/constellation/rewards/DTM.scala:3]: You can remove this interface. Nothing relies on it nor implements it.

File: src/main/scala/org/constellation/rewards/DTM.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  def getAddress: String = address
+
+  def weightByDTM(distribution: Map[String, Double]): Map[String, Double] = {
+    val total = distribution.values.sum
+
+    val reduced = total - perSnapshot
+    val perNode = if (reduced >= 0) reduced / distribution.size else 0d
+
+    val weighted = distribution.transform {
```
Comment (TheMMaciek) [inline src/main/scala/org/constellation/rewards/DTM.scala:23]: Won't `mapValues` be enough?

File: src/main/scala/org/constellation/rewards/SoftStaking.scala [hunk 3]
```diff
@@ -12,20 +15,25 @@ object SoftStaking extends SoftStaking {
 
   def getAddress: String = address
 
-  def weightBySoftStaking(softStakingNodes: Int)(distribution: Map[String, Double]): Map[String, Double] = {
-    val fullNodes = distribution.size
+  def weightBySoftStaking(
+    ignore: Set[String]
+  )(softStakingNodes: Int)(distribution: Map[String, Double]): Map[String, Double] = {
+    val withoutIgnored = distribution.filterNot { case (address, _) => ignore.contains(address) }
```
Comment (TheMMaciek) [inline src/main/scala/org/constellation/rewards/SoftStaking.scala:21]: Can it be implemented just as this - seems like it's semantically the same:
`val withoutIgnored = distribution -- ignore`

File: src/main/scala/org/constellation/rewards/DTM.scala (hunk 1 above)
Comment (marcinwadon) [reply src/main/scala/org/constellation/rewards/DTM.scala:3]: Good catch, I'll use it

File: src/main/scala/org/constellation/rewards/SoftStaking.scala (hunk 3 above)
Comment (marcinwadon) [reply src/main/scala/org/constellation/rewards/SoftStaking.scala:21]: :+1:

File: src/main/scala/org/constellation/rewards/StardustCollective.scala [hunk 4]
```diff
@@ -22,11 +22,14 @@ object StardustCollective extends StardustCollective {
 
   def getAddress(): String = address
 
-  def weightByStardust(distribution: Map[String, Double]): Map[String, Double] = {
-    val stardustWeights = distribution
+  def weightByStardust(ignore: Set[String])(distribution: Map[String, Double]): Map[String, Double] = {
+    val stardustWeights = distribution.filterNot { case (address, reward) => ignore.contains(address) }
```
Comment (TheMMaciek) [inline src/main/scala/org/constellation/rewards/StardustCollective.scala:26]: One more here I guess. You can subtract `ignore`.

**Review by TheMMaciek — APPROVED** [review]

**Pull request merged**